### DISCUSSION TITLE
Force numbers field to require numbers.

### DISCUSSION THREAD

**Discussion by tokkisva** [Budibase/budibase#8685] [Help] [answered]
Hi all,

Sorry if this was already answered, really basic question. Do you know how I can force a numbers field to only contain numbers with validation, or display an error message when alpha characters are entered into a number field? I thought it would be impossible to enter alpha chars, but it does allow it and submits the row anyway and just nulls the values.

I'm not sure how to use validation rules on a number field to cause validation error when alpha characters are entered.

Tried using this rule, but only errors out when you enter a decimal. 
![image](https://user-images.githubusercontent.com/38924262/201748254-9a38e80a-0a4c-4f20-9255-69d6c33f84cc.png)

![image](https://user-images.githubusercontent.com/38924262/201744168-1b67a826-03ff-4473-a0d0-8ba94225ddbf.png)
![image](https://user-images.githubusercontent.com/38924262/201744214-9a14d896-2a84-4dee-9128-6ea3b007247b.png)

Thanks in advance!

**Comment by melohagan** [comment, marked as answer]
Hey @tokkisva 

If you use the number field component then you can only enter numbers:
<img width="239" alt="Screenshot 2022-11-22 at 17 14 12" src="https://user-images.githubusercontent.com/101575380/203378736-b6992045-33ac-4dc6-bd0e-119c89216fdc.png">

Otherwise you can add regex validation to a text field if you really want to only allow numbers: `^[0-9]*$`

**Reply by tokkisva** [reply]
Hi melohagan,

What odd is the field is already a numbers field yet allows alpha characters for some reason. I also cannot set a regex on this field. 
![image](https://user-images.githubusercontent.com/38924262/203406287-a26a1dfc-7875-4bc1-a16a-9e0bfece19eb.png)
![numbers1](https://user-images.githubusercontent.com/38924262/203406096-ffb8cd19-c003-484f-80cf-c8346d1cd335.PNG)
![number2](https://user-images.githubusercontent.com/38924262/203406113-d0b690ac-49df-43d8-995d-8128abf8ae0e.PNG)


However, on mobile devices, it forces the input to numbers. Is this the expected behavior?

**Reply by melohagan** [reply]
That's peculiar. I don't have that issue myself. What operating system and web browser are you using?

**Reply by tokkisva** [reply]
Alright after some messing about, this issue is solely related to Firefox on Windows 10. Chromium based browsers function as intended.

**Reply by melohagan** [reply]
Thanks for checking that @tokkisva 
I've raised a bug here: https://github.com/Budibase/budibase/issues/8779

**Comment by rickydg** [comment]
I have a similar issue on Safari. Form allows text input, but Chrome does not. No regex option under validation.
